src/GNDStk/Node/src/call.hpp

```diff
@@ -0,0 +1,87 @@
+
+// -----------------------------------------------------------------------------
+// Node::operator()
+// -----------------------------------------------------------------------------
+
+// These provide notationally consistent access to the meta() and child()
+// functions, except that the std::string parameter versions of those functions
+// are intentionally not supported here, as it would be indeterminate as to
+// whether we'd want the meta or child (std::string) function. This also
+// reflects the fact that calling our accessor functions with meta_t and
+// child_t parameters - not with std::string parameters - should be preferred.
```

Comment:
The system I'm building will actually be far easier, and much preferred by users. I'll plan to give a talk about this at some point. It takes a bit of a conceptual leap to understand what I'm doing, and how it's going to work. I've done this sort of thing in the past, and it always took some time before I was able to elicit that "aha moment" from other people. Most users won't have to worry about understanding anything advanced, but I'll write clear documentation on what they need to do if they *do* want to do something advanced.

Reply:
If what you are saying is true, then it is a good thing. I'd like to see more about how this is going to be done before you go and spend a lot of time on this. Is this something that you can explain/demonstrate without implementing all of it?

Reply:
Much of the capability is implemented and fully functioning already, and works fabulously. One aspect of it isn't ready quite yet, and can wait until more pressing things are complete. Something more pressing is, for instance, revisiting and completing the JSON capability. I know you'll be happy to see normalize() gone. :-)

Reply:
I'm not sure what you mean by "fully functioning". I don't know that it works until I see tests that compile, run, and pass with our build system. You are making progress on this, but there is still a lot of work to be done.

Reply:
Unfortunately, I've been unable to discern the exact source of this one. You mentioned also in an email, and in person several weeks back, that you haven't been seeing much of the basic tree functionality or tests, which were all up quite some time ago.

All of the material (code and standard tests) regarding this particular discussion have also been up for some time. From everything that's visible from my computer, in my github view at least, it all appears to have uploaded properly. All of it compiles, runs, and passes in tests that I wrote directly in what I believe is the absolutely standard NJOY testing system. I haven't taken the time to become familiar with everything the Catch system provides, but have used it in what seems like a viable way.

How many *.cpp files are you seeing in the Tree test directory? From here, I'm seeing 17. And, if I'm counting correctly, there are 31 *.cpp files in total in all GNDStk/src/GNDStk test/ directories combined.

I've tried to figure out if Github is somehow incorrectly configured, so that the material in question is private, for my view only. If it is, I can't tell.

Have you tried perhaps logging out of github and then back in again, or even just refreshing the page? Also, could there be any issue with metaconfigure's generate.py, or with cmake or some other component going haywire? From here, everything looks good. The only thing I can think of is some sort of broken configuration somewhere, but I can't identify anything here. Much of what I've spoken of was in fact up well before the work-from-home situation, so I can't imagine that anything funny is going on with this computer vs. my onsite office desktop.

At this point I don't think I can spend more time trying to figure this out, or else other more-pressing things will get delayed.

@whaeck What are you seeing in the test/ directories?

Reply:
So, I just pulled down the latest commit to the `feature/tree` branch and don't see any tests for `Node`. There are 17 tests in `src/GNDStk/Tree/test` which is great. I see several .cpp files in test directories as placeholders, but they don't have anything in them. Did you commit these files?

Reply:
I'd mentioned the tree/ tests because those are really the relevant ones in terms of what's spoken of in this particular discussion thread. Node underlies much of tree, and is mostly tested, albeit indirectly, by the tree tests. Nevertheless, I've begun working on direct node tests right now, and will upload sometime soon.

Reply:
With luck, the "conceptual leap" that I spoke of in an early comment, happened this past afternoon.

My keyword stuff should be exercised in at least...

The following four Node test files:

   * GNDStk/src/GNDStk/Node/test/add.test.cpp
   * GNDStk/src/GNDStk/Node/test/call.test.cpp
   * GNDStk/src/GNDStk/Node/test/child.test.cpp
   * GNDStk/src/GNDStk/Node/test/meta.test.cpp

The following three Tree test files:

   * GNDStk/src/GNDStk/Tree/test/call.test.cpp
   * GNDStk/src/GNDStk/Tree/test/child.test.cpp
   * GNDStk/src/GNDStk/Tree/test/meta.test.cpp

The following keyword test file:

   * GNDStk/src/GNDStk/keyword/test/keyword.test.cpp

A bonus feature is a "keyword builder" (look for "keyword.meta" and "keyword.child" here and there) that I didn't have a chance to talk about this afternoon. I'll have to explain it sometime. Basically, it's a shorthand for making meta_t and child_t objects. You made a meta_t object this afternoon when we spoke.

For now, ignore the really wild and woolly looking stuff at the bottom of that keyword.test.cpp file mentined above. I'll need more time to explain that.

Let me know if you *don't* see any of the above eight files. They've been around for a while - at least I see them - and the tests work perfectly.